### ISSUE TITLE
Crowd-trolling shotgun.

### ISSUE THREAD

**Issue by Gregory-Y** [deferred-impact/battledudes-issue-tracker#232]
The triple-barreled saccular-brain-aneurysm-inducing semi-automatic shotgun "Crowd 'troler" trolls and controls crowds by firing a cluster of three .410 bore slugs at a high velocity, which is comparable to sniper rifles. Rather than being made of a dense material for penetrating armor, each slug merely serves to encase a device that, once embedded into a hostile combatant, analyzes the target's brain waves and broadcasts the target's most embarrassing memories to other nearby combatants that are identified as hostile, causing them to develop potentially lethal brain aneurysms due to the immense cringe that they experience.

In the game, once the cluster hits a player, it splits into 3 separate homing projectiles, which then travel towards the nearest enemy players (not the original target), who have engaged the user of the weapon. If a player isn't firing at a Crowd 'troler user, they won't be hit by the secondary projectiles. The projectiles can phase through obstacles and will not stop until they reach the edge of their range, where they disappear.

This means that in theory, it is possible to avoid the Crowd 'troler's secondary projectiles, but it requires the user to run away very quickly and preferably be far away. Even if the enemy is lucky enough to avoid the homing projectile, they must disengage. Thereupon, a lone Crowd 'troler user may succeed in trolling large crowds by forcing them to take a lot of damage or use throwables when attacking.

Cluster stats:
Damage: 10
Falloff: 0.15*
Magazine Capacity: 15
Bullet speed: 115
Fire rate: 250ms
Accuracy: perfect; though this is a triple-barreled shotgun, the projectile is a single bullet because the spread is considered to be negligible, so there's no point in having multiple pellets.
Reload time: 2 seconds.
Range: similar to pump shotgun.

Homing projectile stats:
Damage: 15
Falloff: 0
Bullet speed: 35
Accuracy: they're homing, so perfect I guess.
Range: any player who is M4 range away from the player that got hit by the original cluster can be hit by them after that they disappear.
Noteworthy behaviors: If only a single opponent has engaged the player and they get hit by the Crowd 'troler's cluster, the three homing projectiles will orbit around the player who has been hit, and once another player has managed to hit the user, one of the projectiles will home in on them. Only one projectile from a single cluster may lock onto a player at a time. The rest will remain orbiting until more players engage the Crowd 'troler user or until they disappear after 10 seconds. The projectiles phase through walls and the shield is ineffective. They can also take out tank drivers.

**Comment by deferred-impact** [comment]
Everything sounds good, except the homing projectiles. I identify two issues:
 - somewhat hard to implement, especially since there is no existing code to reuse
 - more importantly, homing projectiles require constant sync between server and client, which results in large bandwidth load

I think the weapon would work just fine with a simple AOE around the victim which grows up to a maximum range. If this AOE hits you then you take the fancy damage. This can even be implemented to support the "orbit around for 10 seconds, wait for a valid target" gameplay.

**Comment by ghost** [comment]
tl;dr

**Comment by deferred-impact** [comment]
> tl;dr

Coming from you, that's funny.

**Comment by ghost** [comment]
Yes, I know, I write long essays full of arguments, but honestly, for a gun description that just looks way too long. Might be a good idea though, I haven't checked yet.

**Comment by deferred-impact** [comment]
I'd rather have an overly long description than a description that fails to explain the key points about an idea

**Comment by ghost** [comment]
Sure. Anyways I'll check this out later, it seems like something interesting, but very long.

**Comment by Gregory-Y** [comment]
I'm probably gonna update the concept in a few days as some things need to be fixed.
